DOCS: general overview of data tiers and roles (#63086)
elastic/elasticsearch: docs/reference/datatiers.asciidoc

[role="xpack"]
[[data-tiers]]
=== Data tiers

> Common data lifecycle management patterns revolve around transitioning the indices

Reviewer: ```suggestion
Common data lifecycle management patterns revolve around transitioning indices
```

through multiple collections of nodes with different hardware characteristics in order

> to fulfil evolving CRUD, search, and aggregation needs as the indices age. The concept

Reviewer: ```suggestion
to fulfil evolving CRUD, search, and aggregation needs as indices age. The concept
```

Each tier formalises specific characteristics and data behaviours.

The node roles that can currently define data tiers are:

* <<data-content-node, data_content>>
* <<data-hot-node, data_hot>>
* <<data-warm-node, data_warm>>
* <<data-cold-node, data_cold>>

The more generic <<data-node, data role>> is not a data tier role, but
it is the default node role if no roles are configured. If a node has the
<<data-node, data>> role we treat the node as if it has all of the tier
roles assigned.

[[content-tier]]
==== Content tier

The content tier is made of one or more nodes that have the <<data-content-node, data_content>>
role. A content tier is designed to store and search user created content. Non-timeseries data
doesn't necessarily follow the hot-warm-cold path. The hardware profiles are quite different to
the <<hot-tier, hot tier>>. User created content prioritises high CPU to support complex
queries and aggregations in a timely manner, as opposed to the <<hot-tier, hot tier>> which
prioritises high IO.
The content data has very long data retention characteristics and from a resiliency perspective
the indices in this tier should be configured to use one or more replicas.

> NOTE: new indices that are not part of <<data-streams, data streams>> will be allocated to the

Reviewer: ```suggestion
NOTE: new indices that are not part of <<data-streams, data streams>> will be automatically allocated to the
```

<<content-tier>>

[[hot-tier]]
==== Hot tier

The hot tier is made of one or more nodes that have the <<data-hot-node, data_hot>> role.
It is the {es} entry point for timeseries data. This tier needs to be fast both for reads
and writes, requiring more hardware resources such as SSD drives. The hot tier is usually
hosting the data from recent days. From a resiliency perspective the indices in this
tier should be configured to use one or more replicas.

> NOTE: new indices that are part of a <<data-streams, data stream>> will be allocated to the

Reviewer: ```suggestion
NOTE: new indices that are part of a <<data-streams, data stream>> will be automatically allocated to the
```

<<hot-tier>>

[[warm-tier]]
==== Warm tier

The warm tier is made of one or more nodes that have the <<data-warm-node, data_warm>> role.

> This tier is where data goes once it is not queried very frequently in the <<hot-tier, hot tier>>
> anymore. It is a medium-fast tier that still allows data updates. The warm tier is usually

Reviewer: ```suggestion
This tier is where data goes once it is not queried as frequently as in the <<hot-tier, hot tier>>.
It is a medium-fast tier that still allows data updates. The warm tier is usually
```

hosting the data from recent weeks. From a resiliency perspective the indices in this
tier should be configured to use one or more replicas.

[[cold-tier]]
==== Cold tier

The cold tier is made of one or more nodes that have the <<data-cold-node, data_cold>> role.
Once the data in the <<warm-tier, warm tier>> is not updated anymore it can transition to the
cold tier. The cold tier is still a responsive query tier but as the data transitions into this

> tier it can be compressed, shrunken, configured to have zero replicas and be backed by

Reviewer: ```suggestion
tier it can be compressed, shrunken, or configured to have zero replicas and be backed by
```

a <<ilm-searchable-snapshot, snapshot>>. The cold tier is usually hosting the data from recent

> months.

Reviewer: ```suggestion
months or years.
```

(since we don't have a frozen tier *yet*)
